### ISSUE TITLE
General cache location

### ISSUE THREAD

**Issue by jthorton** [openforcefield/openff-bespokefit#132]
When running via the CLI users have to provide a working directory else the job is run in a temporary location and all files are removed after meaning that QC calculations and parameters can not be cached. It might be best to have a central location where the redis database is always stored, similar to setting up a qcfractal instance, by default and the directory cmd just overwrites this for the particular run. 

We could then extend the CLI with some management commands which can clean out the cache if users want and restart/ remove calculations from the queue if they have errored.

**Comment by SimonBoothroyd** [comment]
> When running via the CLI users have to provide a working directory else the job is run in a temporary location and all files are removed

This only occurs by default with the `run` command which I think is more meant to be a kind of temporary one off this. I think we should really be encouraging users to use `launch` + `submit` instead for most production use cases which by default creates a local storage directory!

> It might be best to have a central location where the redis database is always stored, similar to setting up a qcfractal instance, by default and the directory cmd just overwrites this for the particular run.

The tricky thing is that because we use the same redis DB for both celery and caching it would mean that if you run `run` once but then decide actually you want to cancel the job because you used the wrong setting, and then launched `run` again the previous job would actually just be resumed, whereas more than likely you would expect a completely new temporary instance to be spawned and only the new job to run.

There are other issues associated with if the user already has a redis instance running for other purposes, then we currently just use that by default and so don't really have control over where it is saved.

We should maybe force users to always spin up their own redis instance to use with bespoke-fit, and raise an error if one is found with the default settings that bepsoke uses?

**Comment by jthorton** [comment]
>I think we should really be encouraging users to use launch + submit instead for most production use cases which by default creates a local storage directory!

Good point maybe we should add a section to the docs detailing this high throughput setup using launch & submit? and probably more details on setting the environment variables used in `Settings`.

>launched run again the previous job would actually just be resumed, whereas more than likely you would expect a completely new temporary instance to be spawned and only the new job to run.

Yeah, that is awkward and led to the idea of having a DB interface similar to QCPortal but I think that might be overkill. Although there could be some use for a minimal interface for example if users do use launch & submit and have a large cache of calculations but make an error during submission it would be useful to have the power to remove the calculation or restart it so the DB does not have to be whipped to start again? 

In this case, it would be best to just print to screen when using `run` that this is temporary and all files will be removed after and that launch & submit is the best option for a long-running DB and I did forget while testing.


>We should maybe force users to always spin up their own redis instance to use with bespoke-fit, and raise an error if one is found with the default settings that bepsoke uses?
 
Could we just tag each instance made by bespokefit in some way so we know if we should use it or not? Like put a simple key in the DB when making it, something like `bespokefit:creator` and it would contain the version used to make it?
